eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -455,6 +455,55 @@ public function loadContentByRemoteId($remoteId, array $languages = null, $versi
         return $content;
     }
 
+    /**
+     * Bulk-load Content items by the list of ContentInfo Value Objects.
+     *
+     * Note: it does not throw exceptions on load, just ignores erroneous Content item.
+     * Moreover, since the method works on pre-loaded ContentInfo list, it is assumed that user is
+     * allowed to access every Content on the list.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\ContentInfo[] $contentInfoList
+     * @param string[] $languages A language priority, filters returned fields and is used as prioritized language code on
+     *                            returned value object. If not given all languages are returned.
+     * @param bool $useAlwaysAvailable Add Main language to \$languages if true (default) and if alwaysAvailable is true
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\Content[] list of Content items with Content Ids as keys
+     */
+    public function loadContentListByContentInfo(
+        array $contentInfoList,
+        array $languages = [],
+        $useAlwaysAvailable = true
+    ) {
+        $contentIds = [];
+        foreach ($contentInfoList as $contentInfo) {
+            $contentIds[] = $contentInfo->id;
+            if (!$useAlwaysAvailable) {
```

Comment:
can also skip if language is all, see note on constant regarding that.

Reply:
I replaced this block with [this logic](https://github.com/ezsystems/ezpublish-kernel/commit/10c2e8ff2bdee08599e5998eea0a2a215700707f#diff-4043c8073a9a8c2593b809f674fafa42R194) you've mentioned in other comment, so this would be redundant. Also changed via fixup 3093853